File: Week_03/G20200343030005/Leetcode_122_001.js
/**
122. 买卖股票的最佳时机 II
*/

/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
  var res = 0;
  var len = prices.length;
  for (var i = 0; i < len - 1; i++) {
    var diff = prices[i + 1] - prices[i];
    if (diff > 0) {
      res += diff;
    }
  }
  return res;
};

File: Week_03/G20200343030001/Leetcode_122_001.java
package Week_03;

public class Leetcode_122_001 {
    public int maxProfit(int[] prices) {
        int profit = 0;

        for (int i = 1; i < prices.length; i++){
            if ( prices[i] > prices[i-1] ) {
                profit += prices[i] - prices[i-1];
            }
        }

        return profit;
    }
}
